slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -96,6 +96,11 @@ ::grpc::Status SLAMServiceImpl::GetPositionNew(
 ::grpc::Status SLAMServiceImpl::GetPointCloudMap(
     ServerContext *context, const GetPointCloudMapRequest *request,
     GetPointCloudMapResponse *response) {
+    // if we have a map for localization, use it
+    if (localization_map_ready) {
```

Comment:
If we're in localization mode but the localization map is not ready, we should return UNAVAILABLE here. Otherwise, we might paint a map (which would contain the new submaps used for localization).

Reply:
kk added a check for if `localization_map_ready == false` and we are in localization mode